core/state_connector_test.go

```diff
@@ -0,0 +1,153 @@
+func (m *mockMessage) Data() []byte                 { return m.data }
+func (m *mockMessage) AccessList() types.AccessList { return m.accessList }
+func (m *mockMessage) IsFake() bool                 { return true }
+
+type mockStateConnectorCaller struct {
+	context  vm.BlockContext
+	testCase string
+	msg      *mockMessage
+}
+
+func (m *mockStateConnectorCaller) Call(caller vm.ContractRef, addr common.Address, input []byte, gas uint64,
+	value *big.Int) (ret []byte, leftOverGas uint64, err error) {
+	switch {
+	case m.testCase == "happy_path" || m.testCase == "finalized_data_error":
+		voter1 := common.BytesToHash([]byte("voter1"))
+		voter2 := common.BytesToHash([]byte("voter2"))
+		merkleRootHash := []byte("some_hash")
+
+		if caller == vm.AccountRef(m.msg.From()) {
+			if addr == common.HexToAddress(GetPrioritisedFTSOContract(nil)) {
+				return []byte("voter"), 0, nil
+			}
+			if addr == common.BytesToAddress([]byte("voter")) {
+				return append([]byte{}, append(voter1.Bytes(), voter2.Bytes()...)...), 0, nil
+			}
+		}
+		if caller == vm.AccountRef(common.BytesToAddress(voter1.Bytes())) ||
+			caller == vm.AccountRef(common.BytesToAddress(voter2.Bytes())) {
+			return merkleRootHash, 0, nil
+		}
+		if caller == vm.AccountRef(stateConnectorCoinbaseSignalAddr(nil, nil)) {
+			if m.context.Coinbase != stateConnectorCoinbaseSignalAddr(nil, nil) {
+				return nil, 0, fmt.Errorf("invalid coinbase address: expected %v, got %v", stateConnectorCoinbaseSignalAddr(nil, nil), m.context.Coinbase)
+			}
+			if m.testCase == "happy_path" {
+				return nil, 0, nil
+			}
+			return nil, 0, errors.New("finalization error")
+		}
+	case m.testCase == "default_attestors_error_1":
+		return nil, 0, errors.New("voter whitelister error")
+	case m.testCase == "default_attestors_error_2":
+		switch addr {
+		case common.HexToAddress(GetPrioritisedFTSOContract(nil)):
+			return []byte("voter"), 0, nil
+		case common.BytesToAddress([]byte("voter")):
+			return nil, 0, errors.New("list price provider error")
+		}
+	}
+	return nil, 0, errors.New("undefined test case")
+}
+
+func (m *mockStateConnectorCaller) WithBlockContext(bc vm.BlockContext) { m.context = bc }
+
+func (m *mockStateConnectorCaller) BlockContext() vm.BlockContext { return m.context }
+
+func TestStateTransition_FinalisePreviousRound(t *testing.T) {
+	t.Run("happy_path", func(t *testing.T) {
+		mockMsg := &mockMessage{
+			from: common.BytesToAddress([]byte("happy_path")),
+		}
+		mockSCC := &mockStateConnectorCaller{
+			testCase: "happy_path",
+			msg:      mockMsg,
+			context: vm.BlockContext{
+				Coinbase: common.BigToAddress(big.NewInt(1000)),
+			},
+		}
+		c := newConnector(mockSCC, mockMsg)
+		currentRoundNumber := []byte("222")
+		err := c.finalisePreviousRound(big.NewInt(10), big.NewInt(10), currentRoundNumber)
+		assert.NoError(t, err)
+		assert.Equal(t, common.BigToAddress(big.NewInt(1000)), mockSCC.context.Coinbase, "coinbase address should be changed to the original address")
+	})
```

Comment:
Why are those tests not parallelized? It seems to me like they don't interact in any way, so they should run in parallel.